Core/src/main/java/org/tribuo/onnx/ONNXUtils.java

```diff
@@ -44,4 +49,29 @@ private ONNXUtils() {}
 
         return builder.build();
     }
+
+    /**
+     * Builds a TensorProto containing the array.
+     * <p>
+     * Downcasts the doubles into floats as ONNX's fp64 support is poor compared to fp32.
```

Comment:
Do we want to future-proof this a bit by supporting both fp64 and fp32 now, but defaulting to downcasting? That way we won't need to change the interface if support gets better or we care more about it in the future?

Reply:
I can add an overload with a boolean flag and have this one call it with `downcast` to true.

Reply:
Done.